common/txmgr/confirmer.go

```diff
@@ -133,14 +129,14 @@ type Confirmer[
 	ks               txmgrtypes.KeyStore[ADDR, CHAIN_ID, SEQ]
 	enabledAddresses []ADDR
 
-	mb        *mailbox.Mailbox[HEAD]
-	stopCh    services.StopChan
-	wg        sync.WaitGroup
-	initSync  sync.Mutex
-	isStarted bool
+	mb           *mailbox.Mailbox[HEAD]
+	stopCh       services.StopChan
+	wg           sync.WaitGroup
+	initSync     sync.Mutex
+	isStarted    bool
+	isReceiptNil func(R) bool
 
-	nConsecutiveBlocksChainTooShort int
-	isReceiptNil                    func(R) bool
+	confirmerHeadTracker confirmerHeadTracker[HEAD, BLOCK_HASH]
```

Comment:
nit: I think the field itself we can just name headTracker for simplicity since it's inside the Confirmer
```suggestion
	headTracker confirmerHeadTracker[HEAD, BLOCK_HASH]
```